specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,207 @@
+    _webViewEnvironment = await CoreWebView2Environment.CreateAsync(
+            options: customOptions);
+    if (_webViewEnvironment != null)
+    {
+        CheckChannel(_webviewEnvironment.BrowserVersionString());
+    }
+}
+```
+```cpp
+void AppWindow::InitializeWebViewEnvironment()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    std::vector<COREWEBVIEW2_RELEASE_CHANNEL> channels =
+    {
+        COREWEBVIEW2_RELEASE_CHANNEL_BETA,
+        COREWEBVIEW2_RELEASE_CHANNEL_STABLE
+    };
+    CHECK_FAILURE(options->SetReleaseChannelPreferenceSet(
+        channels.size(), channels.data()));
+    // If the loader is unable to find a valid installation from the release
+    // channel preference set, environment creation will fail with
+    // HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND).
+    HRESULT hr = CreateCoreWebView2EnvironmentWithOptions(
+        nullptr, m_userDataFolder.c_str(), options.Get(),
+        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+            this, &AppWindow::OnCreateEnvironmentCompleted)
+            .Get());
+}
+void AppWindow::OnCreateEnvironmentCompleted(
+    HRESULT result, ICoreWebView2Environment* environment)
+{
+    if (environment)
+    {
+        wil::unique_cotaskmem_string version;
+        CHECK_FAILURE(environment->get_BrowserVersionString(&version));
+        CheckChannel(version.get());
+    }
+}
+```
+# API Details
+```
+typedef enum COREWEBVIEW2_RELEASE_CHANNEL {
+  /// The stable WebView2 Runtime that is released every 4 weeks.
+  COREWEBVIEW2_RELEASE_CHANNEL_STABLE,
+  /// The Beta release channel that is released every 4 weeks, a week before the
+  /// stable release.
+  COREWEBVIEW2_RELEASE_CHANNEL_BETA,
+  /// The Dev release channel that is released weekly.
+  COREWEBVIEW2_RELEASE_CHANNEL_DEV,
+  /// The Canary release channel that is released daily.
+  COREWEBVIEW2_RELEASE_CHANNEL_CANARY,
+} COREWEBVIEW2_RELEASE_CHANNEL;
+
+/// Additional options used to create the WebView2 Environment that support
+/// specifying the `ReleaseChannelPreferenceSet`.
+[uuid(47ac856e-a726-4e04-b36b-f58dafb39e38), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+
+  /// Gets the `ReleaseChannelPreferenceSet`. The memory pointed to by *channels
+  /// is owned by the `ICoreWebView2EnvironmentOptions` object and there
+  /// is no need to release it. The memory also cannot be used after the
+  /// `ICoreWebView2EnvironmentOptions` object is released.
+  HRESULT GetReleaseChannelPreferenceSet(
+      [out] UINT32* count,
+      [out] COREWEBVIEW2_RELEASE_CHANNEL** channels);
+
+  /// Sets the `ReleaseChannelPreferenceSet`. By default, the WebView2 loader
+  /// searches for channels from most to least stable, using the first channel
+  /// found on the device. When a `ReleaseChannelPreferenceSet` is provided, the
+  /// channel search order is reversed and the loader searches for the provided
```

Comment:
That makes sense to me. Would this also apply to the environment variable and reg key? So they would have to set two separate overrides, like the current ReleaseChannelPreferenceSet (or a new one that matches the property name) along with ReleaseChannelPreferenceSet?

Reply:
I would expect so. We generally have env var + regkey overrides per loader option. The releasechannelpreference regkey+env var is a weird one out because it was at one point also in the API but then the API was removed. I'd be OK not adding a duplicate regkey+env var for releasechannelpreference and just continuing on with the weirdly named existing one, but also would be OK with adding a duplicate with a 'good' name